modules/apps/transfer/types/transfer_authorization_test.go

```diff
@@ -1,132 +1,161 @@
-	toAddr         = sdk.AccAddress("_______to________")
-	timeoutHeight  = clienttypes.NewHeight(0, 10)
-)
+func (suite *TypesTestSuite) TestTransferAuthorizationAccept() {
+	var (
+		msgTransfer   types.MsgTransfer
+		transferAuthz types.TransferAuthorization
+	)
 
-func TestTransferAuthorization(t *testing.T) {
-	app := simapp.Setup(t, false)
-	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
-	allocation := types.Allocation{
-		SourcePort:    sourcePort,
-		SourceChannel: sourceChannel,
-		SpendLimit:    coins1000,
-		AllowList:     []string{toAddr.String()},
-	}
-	authorization := types.NewTransferAuthorization(allocation)
-
-	t.Log("verify authorization returns valid method name")
-	require.Equal(t, authorization.MsgTypeURL(), "/ibc.applications.transfer.v1.MsgTransfer")
-	require.NoError(t, authorization.ValidateBasic())
-	transfer := types.NewMsgTransfer(sourcePort, sourceChannel, coin1000, fromAddr.String(), toAddr.String(), timeoutHeight, 0, "")
-	require.NoError(t, authorization.ValidateBasic())
-
-	t.Log("verify updated authorization returns nil")
-	resp, err := authorization.Accept(ctx, transfer)
-	require.NoError(t, err)
-	require.True(t, resp.Delete)
-	require.Nil(t, resp.Updated)
-
-	t.Log("verify updated authorization returns remaining spent limit")
-	authorization = types.NewTransferAuthorization(allocation)
-	require.Equal(t, authorization.MsgTypeURL(), "/ibc.applications.transfer.v1.MsgTransfer")
-	require.NoError(t, authorization.ValidateBasic())
-	transfer = types.NewMsgTransfer(sourcePort, sourceChannel, coin500, fromAddr.String(), toAddr.String(), timeoutHeight, 0, "")
-	require.NoError(t, authorization.ValidateBasic())
-	resp, err = authorization.Accept(ctx, transfer)
-	require.NoError(t, err)
-	require.False(t, resp.Delete)
-	require.NotNil(t, resp.Updated)
-
-	allocation = types.Allocation{
-		SourcePort:    sourcePort,
-		SourceChannel: sourceChannel,
-		SpendLimit:    coins500,
-		AllowList:     []string{toAddr.String()},
-	}
-	sendAuth := types.NewTransferAuthorization(allocation)
-	require.Equal(t, sendAuth.String(), resp.Updated.String())
-
-	t.Log("expect updated authorization nil after spending remaining amount")
-	resp, err = resp.Updated.Accept(ctx, transfer)
-	require.NoError(t, err)
-	require.True(t, resp.Delete)
-	require.Nil(t, resp.Updated)
-
-	t.Log("expect error when spend limit for specific port and channel is not set")
-	allocation = types.Allocation{
-		SourcePort:    sourcePort,
-		SourceChannel: sourceChannel,
-		SpendLimit:    coins1000,
-		AllowList:     []string{toAddr.String()},
-	}
-	authorization = types.NewTransferAuthorization(allocation)
-	transfer = types.NewMsgTransfer(sourcePort2, sourceChannel2, coin500, fromAddr.String(), toAddr.String(), timeoutHeight, 0, "")
-	_, err = authorization.Accept(ctx, transfer)
-	require.Error(t, err)
+	testCases := []struct {
+		name      string
+		malleate  func()
+		expPass   bool
+		expResult func(res authz.AcceptResponse)
```

Comment:
[nit] the name `expResult` to me indicates that a function should return the result which will later be used in some sort of assertion, but this function is actually doing the assertions. Maybe `assertResult` or similar is more appropriate WDYT?

Reply:
As a follow up, if this function is renamed, we no longer really need the boolean `expPass` as that assertion could be done in the function too, in this case maybe the method signature could change to `func(res authz.AcceptResponse, err error)`

Reply:
I've applied the suggested changes here, but I think I'll leave the malleate func as is for now. I agree it would be good to limit the scope of what can be mutated in the malleate function but for now I'll just keep it as is to be consistent with other tests